lib/utils/testlog/log.go

```diff
@@ -0,0 +1,30 @@
+// Package testlog provides custom loggers for use in tests.
+package testlog
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+)
+
+// FailureOnly returns a logger that only prints the logs to STDERR when the
+// test fails.
+func FailureOnly(t *testing.T) *logrus.Entry {
```

Comment:
we have `utils.InitLoggerForTests(testing.Verbose())` Does it make sense to use `FailureOnly` logger in that function too by default?

Reply:
`FailureOnly` depends on per-test state (`testing.T`)
`utils.InitLoggerForTests` is inherently global and doesn't have access to per-test info.
I can't think of a way to implement something similar to `utils.InitLoggerForTests` using `FailureOnly`